src/main/java/org/apache/commons/collections4/map/DefaultedMap.java

```diff
@@ -105,7 +105,8 @@
      * @param map  the map to decorate, must not be null
      * @param factory  the factory to use to create entries, must not be null
      * @return a new defaulting map
-     * @throws NullPointerException if map or factory is null
+     * @throws NullPointerException if map is null
+     * @throws IllegalArgumentException if factory is null
```

Comment:
This is indeed a judgment of NullPointerException. It is recommended to modify IllegalArgumentException to NullPointerException, and to supplement the test cases for it.

Reply:
Agreed. I changed to NullPointerException and added test cases for it.